django/core/management/commands/makemessages.py

```diff
@@ -351,9 +351,10 @@ def handle(self, *args, **options):
         else:
             if self.settings_available:
                 self.locale_paths.extend(settings.LOCALE_PATHS)
-            # Allow to run makemessages inside an app dir
-            if os.path.isdir('locale'):
+            # Allow running makemessages inside an app dir
+            if os.path.isdir('locale') or not self.settings_available:
```

Comment:
This change is not tested and I'm not sure if it's excepted. I would focus on improving an error message.

Reply:
Yeah, this was meant to be a sort of auto resolution of the error. But I agree, we should focus first on improving the error message.